### DISCUSSION TITLE
Strict LABEL use

### DISCUSSION THREAD

**Discussion by struthio** [moosefs/moosefs#411] [Q&A]
hi,

I have question (if such configuration is possible in MooseFS).
Consider such configuration:
Chunkserver1 - LABEL A
Chunkserver2 - LABEL B
Chunkserver3 - LABEL C

Two storage classes:
sclass1 - A,B
sclass2 - A,C

Two directories on MooseFS
Test1 - Assigned sclass1
Test2 - Assigned sclass2

Everything is replicated according to rules. Now when I switch off Chunkserver2, directory Test1 is under-goal and _starts automatic replication_ to Chunkserver3.

I would like to avoid this replication, if there is no chunkserver with label B, then I would like Test1 to be under-goal until somebody adds new "B" chunkserver.

Is such configuration possible in MooseFS ?

**Comment by struthio** [comment]
Ok I see that I am using old version 3.0.109 and some changes was added since then [https://github.com/moosefs/moosefs/commit/52371d082b0946b2de00dcf64b92b92284e69e5a](https://github.com/moosefs/moosefs/commit/52371d082b0946b2de00dcf64b92b92284e69e5a)

**Comment by chogata** [comment]
Yes, you should definitely upgrade. In current version of MooseFS the strict mode will behave just like you want - if you switch off the chunkserver with label B, your chunks in class sclass1 will remain undergoal.
